xchange-lgo/src/main/java/org/knowm/xchange/lgo/service/LgoTradeService.java

```diff
@@ -131,61 +116,69 @@ private LgoProduct getProduct(CurrencyPair currencyPair) {
 
   @Override
   public String placeLimitOrder(LimitOrder limitOrder) throws IOException {
-    LgoPlaceOrder lgoOrder = LgoAdapters.adaptLimitOrder(limitOrder);
-    return placeOrder(lgoOrder);
+    try {
+      LgoUnencryptedOrder lgoOrder = LgoAdapters.adaptUnencryptedLimitOrder(limitOrder);
+      return placeLgoUnencryptedOrder(lgoOrder);
+    } catch (LgoException e) {
+      throw LgoErrorAdapter.adapt(e);
+    }
   }
 
-  @Override
-  public String placeMarketOrder(MarketOrder marketOrder) throws IOException {
-    LgoPlaceOrder lgoOrder = LgoAdapters.adaptMarketOrder(marketOrder);
-    return placeOrder(lgoOrder);
+  /** Place a limit order encrypting it's content. */
+  public String placeEncryptedLimitOrder(LimitOrder limitOrder) throws IOException {
+    LgoPlaceOrder lgoOrder = LgoAdapters.adaptLimitOrder(limitOrder);
+    return placeEncryptedOrder(lgoOrder);
   }
 
   @Override
-  public boolean cancelOrder(String orderId) throws IOException {
-    LgoPlaceOrder lgoOrder = LgoAdapters.adaptCancelOrder(orderId, new Date());
-    placeOrder(lgoOrder);
-    return true;
-  }
-
-  private String placeOrder(LgoPlaceOrder lgoOrder) throws IOException {
+  public String placeMarketOrder(MarketOrder marketOrder) throws IOException {
```

Comment:
why is sending the unencrypted order by default better?

Reply:
Good question. Encryption is about anti front running. It's a way for LGO to prove that no manipulation has occurred on a given batch. We have some explanations [here](https://lgono.de/)

The unencrypted support was added a few months ago, so now it's a matter of choice for a market participant to activate or not anti frontrunning protection. Of course, that raises the question of which version should be the default here, given the fact that LGO, as a regulated exchange, is also audited by more regular means. 
Since encryption comes with a cost, if you are not aware of this feature, that means that you would always be a little bit slower to place your orders than customer not using it. 

Thinking out loud, maybe a better implementation here to avoid custom methods would be to add a configuration parameter for LgoExchange, and check its value here.

Reply:
I agree, having a exchange specific parameter in exchange specification that would determine if we send encrypted orders or not would be better

Reply:
Done.